packages/expo-store-review/ios/EXStoreReview/EXStoreReviewModule.m

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <EXStoreReview/EXStoreReviewModule.h>
+#import <StoreKit/SKStoreReviewController.h>
+
+@implementation EXStoreReviewModule
+
+UM_EXPORT_MODULE(ExpoStoreReview);
+
+- (NSDictionary *)constantsToExport
+{
+  BOOL isSupported;
+  if (@available(iOS 10.3, *)) {
+    isSupported = [SKStoreReviewController class] ? YES : NO;
+  } else {
+    isSupported = NO;
+  }
+  return @{ @"isSupported": @(isSupported) };
+}
+
+UM_EXPORT_METHOD_AS(requestReview,
+                    resolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  if (@available(iOS 10.3, *)) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [SKStoreReviewController requestReview];
+    });
+    resolve(nil);
```

Comment:
How about moving this `resolve(nil)` to inside `dispatch_async`? This way we will resolve the Promise once the controller is actually shown.